src/qirlightning/LightningQuantum.cc

```diff
@@ -0,0 +1,200 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other QIR-EE developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//---------------------------------------------------------------------------//
+//! \file qirlightning/LightningQuantum.cc
+//---------------------------------------------------------------------------//
+
+#include "LightningQuantum.hh"
+
+#include <algorithm>
+#include <iostream>
+#include <optional>
+#include <random>
+#include <stdexcept>
+#include <thread>
+#include <utility>
+#include <dlfcn.h>
+
+#include "qiree/Assert.hh"
+
+extern "C" Catalyst::Runtime::QuantumDevice*
+GenericDeviceFactory(char const* kwargs);
+namespace qiree
+{
+using namespace Catalyst::Runtime;
+
+//---------------------------------------------------------------------------//
+/*!
+ * Initialize the Lightning simulator
+ */
+LightningQuantum::LightningQuantum(std::ostream& os, unsigned long int seed)
+    : output_(os), seed_(seed)
+{
+    std::string rtd_lib = RTDLIB;
+    std::string rtd_device = RTDDEVICE;
+    std::string kwargs = {};
+    auto rtld_flags = RTLD_LAZY | RTLD_NODELETE;
+    rtd_dylib_handler = dlopen(rtd_lib.c_str(), rtld_flags);
+
+    if (!rtd_dylib_handler)
+    {
+        throw std::runtime_error("Failed to load library: " + rtd_lib);
+    }
```

Comment:
QIREE has macros that simplify this and throw a specialized runtime subclass:
```suggestion
    QIREE_VALIDATE(rtd_dylib_handler,
        << "failed to load Lightning runtime library '" << RTDLIB << "'");
```